### DISCUSSION TITLE
is it possible build an "Angular Style DI" library using Reflectable?

### DISCUSSION THREAD

**Discussion by dinbtechit** [google/reflectable.dart#297] [General]
is it possible build an Angular-style DI (Dependency Injection) library using Reflectable? Looking to build a DI library that would eliminate the need to generate code (using `build_runner`) at the development time.

**For example:** The usage of the DI library will be as shown below

```dart
@Injectable 
class ServiceA {
 
}

@Widget
class MyWidget extends StatefulWidget {
  ServiceA serviceA  
  MyWidget(this.serviceA);         // <----<< At the Deployment time I would like to automatically inject an object of serviceA.
   
}
```

**Comment by sigurdm** [comment]
Something like this is most probably possible, however:

> Looking to build a DI library that would eliminate the need to generate code (using build_runner) at the development time.

Reflectable itself needs to generate code, usually done with build_runner.

**Reply by dinbtechit** [reply]
Thank you for the response! 

You're so right.. I realized that after doing some small experiments with `Reflectable`. I thought reflectable works dynamically at runtime but apparently not. 

Too bad without `dart:mirrors` support in Flutter. It's kinda hard to write nifty/intuitive frameworks for dart. Maybe I am spoiled with Angular Typescript and Spring framework. lol
